tutorials/time-series-modeling/Lab_05.md

```diff
@@ -0,0 +1,392 @@
+
+```javascript
+// Use this function to add variables for NDVI, time and a constant
+// to Landsat 8 imagery.
+var addVariables = function(image) {
+  // Compute time in fractional years since the epoch.
+  var date = ee.Date(image.get(timeField));
+  var years = date.difference(ee.Date('1970-01-01'), 'year');
+  // Return the image with the added bands.
+  return image
+  // Add an NDVI band.
+  .addBands(image.normalizedDifference(['B5', 'B4']).rename('NDVI'))
+  // Add a time band.
+  .addBands(ee.Image(years).rename('t'))
+  .float()
+  // Add a constant band.
+  .addBands(ee.Image.constant(1));
+};
+// Remove clouds, add variables and filter to the area of interest.
+var filteredLandsat = landsat_8_sr
+.filterBounds(roi)
+.map(maskL8sr)
+.map(addVariables);
+```
+
+To visualize the data, we will export a chart at the location of interest. We will add a linear trend line for reference. 
+
+```javascript
+// Plot a time series of NDVI at a single location.
+var l8Chart = ui.Chart.image.series(filteredLandsat.select('NDVI'), roi)
+  .setChartType('ScatterChart')
+  .setOptions({
+   title: 'Landsat 8 NDVI Time Series at ROI',
+   trendlines: {0: {
+				color: 'CC0000'
+   }},
+   lineWidth: 1,
+   pointSize: 3,
+  });
+print(l8Chart);
+```
+
+You can click on the 'export' button next to the chart to view an interactive chart. Scroll over some of the data points and look at the relationships between the data. A line connecting two dots means that they are sequential data points (notice that there are relatively few sequential points). We can also see that there are relatively large jumps in the data, with an upward climb somewhere between March and late April, and a descent in late August. Each year is slightly different, but we can surmise that this is due to seasonal rains in the spring and leaves dying off in the fall. Finally, the general trend is downward, although the February 2021 datapoint might have significant leverage on the trend. 
+
+![Linear Trend](./im/im_05_03.png)
+
+
+### Linear Modeling of Time
+Lots of interesting analyses can be done to time series by harnessing the `linearRegression()` [reducer](https://developers.google.com/earth-engine/api_docs#eereducerlinearregression). To estimate linear trends over time, consider the following linear model, where $\epsilon_t$ is a random error:
+
+$$y = \beta_0 + \beta_1X_1 + ... + \beta_nX_n + \epsilon_t$$
+
+This is the model behind the trendline added to the chart you just created. We can use this model to detrend our data (explain the upward or downward movement of the data by subtracting observed values from the fitted model values). For now, the goal is to discover the values of the beta coefficients.
+
+To fit this trend model to the Landsat-based NDVI series using Ordinary Least Squares ([OLS](https://statisticsbyjim.com/glossary/ordinary-least-squares/)), use the `linearRegression()` reducer:                                                                
```

Comment:
Regarding the URL for OLS, I wonder if you can find a resource that has less advertisements. I looked briefly, but didn't find something that I was happy with. Wikipedia is always a good option, though in this case they don't provide just a simple figure to illustrate the concept--I think it would be a fine link, though.

Reply:
https://setosa.io/ev/ordinary-least-squares-regression/
I thought this was an interesting site, with visual explanation on the concept.

Reply:
Nice, that's cool!